### PR TITLE
Meson i2c (DRAFT)

### PR DESCRIPTION
**Pull request by omeh-a**
Single-server multi-interface i2c driver for the ODROID C4.

### CODE REVIEW

File: drivers/i2c/odroidc4/i2c-client.c
Comment (wom-bat) [inline drivers/i2c/odroidc4/i2c-client.c:1]: This should be in the examples directory ( and should actually do something(

File: drivers/i2c/odroidc4/i2c-sel4cp.h
Comment (wom-bat) [inline drivers/i2c/odroidc4/i2c-sel4cp.h:1]: Needs microkit rebranding

**Comment by wom-bat** [comment]
There's a heap that should be general to any i2c host interface that's tucked into odroidc4 directories.

**Comment by Ivan-Velickovic** [comment]
Will re-open to merge into `lionsos` later on.

**Pull request closed without merging**